### PR TITLE
fix(hummock): fix gc delete keys incorrectly

### PR DESCRIPTION
**Pull request by Little-Wallace**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Fix bug cause by https://github.com/risingwavelabs/risingwave/pull/12534

For LSM tree architecture,we must remove all tombstone at last level, because we do not know whether there are still some put keys in lower level which shall be deleted by this tombstone.
If we remove tombstone in an upper level, those deleted put keys may be saw by user  again and it will cause inconsistent in state compute.

This bug may be related to https://github.com/risingwavelabs/risingwave/issues/14592 and https://github.com/risingwavelabs/risingwave/issues/14493

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by hzxa21 — APPROVED** [review]

File: src/stream/src/executor/over_window/general.rs [hunk 1]
```diff
@@ -382,7 +386,16 @@ impl<S: StateStore> OverWindowExecutor<S> {
...
-                            _ => panic!("other cases should not exist"),
+                            _ => {
+                                let vnode =
+                                    this.state_table.compute_vnode_by_pk(&key.pk);
+                                let raw_key = serialize_pk_with_vnode(
+                                    &key.pk,
+                                    this.state_table.pk_serde(),
+                                    vnode,
+                                );
+                                panic!("other cases should not exist. raw_key: {:?}, existed: {:?}, new: {:?}", raw_key, tp1, tp2);
```
Comment (Li0k) [inline src/stream/src/executor/over_window/general.rs:397]: Is it expected ?

**Review by Li0k — APPROVED** [review]
LGTM,thanks

File: src/stream/src/executor/over_window/general.rs (hunk 1 above)
Comment (Little-Wallace) [reply src/stream/src/executor/over_window/general.rs:397]: I think it is necessary to output content of wrong key in panic

File: src/stream/src/executor/over_window/general.rs [hunk 2]
```diff
@@ -382,7 +387,15 @@ impl<S: StateStore> OverWindowExecutor<S> {
...
                             }
-                            _ => panic!("other cases should not exist"),
+                            _ => {
+                                let vnode = this.state_table.compute_vnode_by_pk(&key.pk);
+                                let raw_key = serialize_pk_with_vnode(
+                                    &key.pk,
+                                    this.state_table.pk_serde(),
+                                    vnode,
+                                );
+                                panic!("other cases should not exist. raw_key: {:?}, existed: {:?}, new: {:?}", raw_key, tp1, tp2);
```
Comment (stdrc) [inline src/stream/src/executor/over_window/general.rs:397]: Feeling a little bit weird to print raw table key in executor. But it looks acceptable for error avoidance. I would prefer to move the `let tp1 = existed.to_record_type(); ...` part and the `use` statement to the panic branch to make the debug things stay together.

Comment (Little-Wallace) [reply src/stream/src/executor/over_window/general.rs]: fix

**Review by stdrc — APPROVED** [review]

**Comment by Little-Wallace** [comment]
I run [longevity-test](https://buildkite.com/risingwave-test/longevity-test) for this branch for 6 hours and it does not cause any panic

**Pull request merged**